### PR TITLE
People search

### PR DESCRIPTION
**Pull request by mattss**
This mainly introduces the 'People' search tab, with both zcatalog and solr backend support.

Also:
- Moves the avatar generation code out of the api
- Robot tests for other search tabs

### CODE REVIEW

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/288/
Test PASSed.

File: src/ploneintranet/api/userprofile.py [hunk 1]
```diff
@@ -145,47 +146,20 @@ def create(
...
-        )
-    except TypeError:
-        # No image found
-        return None
-
-    if scale is not None:
-        return scale.url
-    else:
-        return None
+    return '{0}/@@avatar.jpg'.format(profile.absolute_url())
```
Comment (gyst) [inline src/ploneintranet/api/userprofile.py:165]: Thanks for moving that out of the API into a view :-)

**Comment by gyst** [comment]
There's a bug: if you do an empty people search you get a traceback.

Separately there's the issue that it's highly confusing if you search and get no results. Because you're not logged in :-). The search view should be protected by ploneintranet.viewsite, like
https://github.com/ploneintranet/ploneintranet/blob/master/src/ploneintranet/layout/browser/configure.zcml#L27

You will need to `<include package="ploneintranet.layout" file="permissions.zcml" />` for that.

**Comment by ghost** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://jenkins.ploneintranet.net/job/Plone.Intranet.Pull.Request.Builder/292/
Test PASSed.

**Comment by mattss** [comment]
@gyst Thanks for the review - issues resolved.

**Pull request merged**